src/main/java/com/google/cloud/tools/maven/cloudsdk/CloudSdkAppEngineFactory.java

```diff
@@ -130,12 +133,16 @@ static CloudSdk buildCloudSdk(
         return cloudSdk;
       } else {
         // we need to use a managed cloud sdk
+        List<SdkComponent> requiredComponents = new ArrayList<>();
+        if (requiresAppEngineComponents) {
+          requiredComponents.add(SdkComponent.APP_ENGINE_JAVA);
```

Comment:
There's no prod code that requests to install `SdkComponent.BETA`. Is the API change necessary?

Reply:
I think it's okay, I like this pattern more, rather than passing around some boolean all over the place.